### ISSUE TITLE
False-positive `ArgumentTypeCoercion` on `Builder::each(fn (TModel $m) => ...)`

### ISSUE THREAD

**Issue by alies-dev** [psalm/laravel-psalm-plugin#793]
## Summary

`Illuminate\Database\Eloquent\Builder::each()` raises a false-positive `ArgumentTypeCoercion` when the user passes a closure that (a) narrows the model param to a concrete subclass, or (b) omits the second `int` parameter.

## Reproduction

```php
User::query()->where('is_admin', true)->each(static function (User $user): void {
    $user->syncRoles(Role::ADMIN);
});
```

Psalm error:

```
ArgumentTypeCoercion — Argument 1 of Illuminate\Database\Eloquent\Builder::each
expects impure-callable(Illuminate\Database\Eloquent\Model, int):mixed,
but parent type impure-Closure(App\Models\User):void provided
```

## Root cause

Laravel declares `each()` on the `Illuminate\Database\Concerns\BuildsQueries` trait with `@param callable(TValue, int): mixed` (`vendor/laravel/framework/src/Illuminate/Database/Concerns/BuildsQueries.php:105`). The plugin does not override this for `Illuminate\Database\Eloquent\Builder`, so:

1. **`TValue` does not bind to the Builder's `TModel`** — it resolves to `Model`, the trait's default. A closure typed `function (User $user)` is narrower than `Model`, violating callable contravariance.
2. **Arity mismatch** — Laravel invokes `$callback($value, $key)` at runtime (two args), but PHP lets callers ignore the extra arg. Psalm's callable check does not; it rejects a 1-param closure where a 2-param callable is declared.

## Proposed fix

Add an Eloquent-`Builder`-specific `each()` stub to the plugin, binding `TValue` to `TModel` and widening the callable arity, e.g.:

```php
/**
 * @param  (callable(TModel, int): mixed)|(callable(TModel): mixed)  $callback
 * @param  int  $count
 * @return bool
 */
public function each(callable $callback, $count = 1000) {}
```

The two-arg variant matches the runtime contract; the one-arg variant accommodates the common user pattern (and is safe at runtime — PHP discards extra args).

Same treatment likely needed for the `Relation`/`BelongsToMany`/`HasOneOrManyThrough` overrides of `each()` at `vendor/laravel/framework/src/Illuminate/Database/Eloquent/Relations/BelongsToMany.php:1132` and `…/HasOneOrManyThrough.php:618`.

## Real-world impact

Reported in a migration calling `User::query()->…->each(fn (User $u) => …)`. This is the canonical pattern for running a one-off operation over matching rows and currently requires `// @phpstan-ignore-next-line` or equivalent suppression.

## Related

Not a dupe of #776 (that fix targeted the `where`-family closures). #774 is a related "widen-to-match-runtime" precedent.
